common/flatpak-context.c

```diff
@@ -1669,15 +1674,13 @@ flatpak_context_save_metadata (FlatpakContext *context,
     }
 
   g_key_file_remove_group (metakey, FLATPAK_METADATA_GROUP_ENVIRONMENT, NULL);
-  g_hash_table_iter_init (&iter, context->env_vars);
-  while (g_hash_table_iter_next (&iter, &key, &value))
+  for (int i = 0; i < context->env_vars->len; i++)
     {
-      g_key_file_set_string (metakey,
-                             FLATPAK_METADATA_GROUP_ENVIRONMENT,
-                             (char *) key, (char *) value);
+      char *key = g_ptr_array_index (context->env_vars, i);
+      char *value = g_ptr_array_index (context->env_vals, i);
+      g_key_file_set_string (metakey, FLATPAK_METADATA_GROUP_ENVIRONMENT, key, value);
     }
```

Comment:
Is this guaranteed to keep the order?

Reply:
That is true, but ignoring conflicts, does adding a new key always add it at the end of the group?

Reply:
I tried to make sure thats the case, and my tests seems to confirm that it is